bouncer/shared/gaslimit_ccm.ts

```diff
@@ -119,130 +122,59 @@ async function trackGasLimitSwap(
+
+  // Wait for the swap to complete
+  const outputAmount = (
+    await observeEvent('swapping:SwapExecuted', {
+      test: (event) => Number(event.data.swapRequestId.replaceAll(',', '')) === swapRequestId,
+      historicCheckBlocks: CHECK_PAST_BLOCKS_FOR_EVENTS,
+    }).event
+  ).data.outputAmount;
+
+  // Find all of the swap events by looking at the event history
+  const egressId = (
+    await observeEvent('swapping:SwapEgressScheduled', {
+      test: (event) => Number(event.data.swapRequestId.replaceAll(',', '')) === swapRequestId,
+      historicCheckBlocks: CHECK_PAST_BLOCKS_FOR_EVENTS,
+    }).event
+  ).data.egressId as EgressId;
+
+  const broadcastId = (
+    await observeEvent(`${destChain.toLowerCase()}IngressEgress:CcmBroadcastRequested`, {
       test: (event) => {
-        broadcastIdToTxPayload[event.data.broadcastId] = event.data.transactionPayload;
-        return false;
+        console.log(`event.data.egressId ${event.data.egressId} / ${egressId}`);
+        return event.data.egressId[0] === egressId[0] && event.data.egressId[1] === egressId[1];
       },
-      abortable: true,
-    },
-  );
-
-  await send(sourceAsset, depositAddress);
+      historicCheckBlocks: CHECK_PAST_BLOCKS_FOR_EVENTS,
+    }).event
+  ).data.broadcastId;
 
-  const swapRequestId = (await swapRequestedHandle).data.swapRequestId;
-
-  while (
-    !(
-      swapRequestId in reqIdToPrincipalSwapId &&
-      reqIdToPrincipalSwapId[swapRequestId] in swapIdToEgressAmount &&
-      swapRequestId in swapRequestIdToEgressId &&
-      swapRequestIdToEgressId[swapRequestId] in egressIdToBroadcastId &&
-      egressIdToBroadcastId[swapRequestIdToEgressId[swapRequestId]] in broadcastIdToTxPayload
-    )
-  ) {
-    await sleep(3000);
-  }
-
-  const egressId = swapRequestIdToEgressId[swapRequestId];
-  const broadcastId = egressIdToBroadcastId[egressId];
-
-  console.log(`${tag} Swap events found`);
-
-  swapExecutedHandle.stop();
-  swapEgressHandle.stop();
-  ccmBroadcastHandle.stop();
-  broadcastRequesthandle.stop();
-  swapScheduledHandle.stop();
-
-  await Promise.all([
-    swapExecutedHandle.event,
-    swapEgressHandle.event,
-    ccmBroadcastHandle.event,
-    broadcastRequesthandle.event,
-    swapScheduledHandle.event,
-  ]);
-
-  let egressBudgetAmount;
+  const txPayload = (
+    await observeEvent(`${destChain.toLowerCase()}Broadcaster:TransactionBroadcastRequest`, {
+      test: (event) => event.data.broadcastId === broadcastId,
+      historicCheckBlocks: CHECK_PAST_BLOCKS_FOR_EVENTS,
+    }).event
+  ).data.transactionPayload;
 
-  if (chainGasAsset(destChain) === sourceAsset) {
-    egressBudgetAmount = messageMetadata.gasBudget;
-  } else {
-    const gasSwapId = reqIdToGasSwapId[swapRequestId];
-    egressBudgetAmount = Number(swapIdToEgressAmount[gasSwapId].replace(/,/g, ''));
-  }
+  const egressBudgetAmount =
+    chainGasAsset(destChain as Chain) === sourceAsset ? messageMetadata.gasBudget : outputAmount;
```

Comment:
Maybe I'm mistaken here, since swap events have recently changed, but I think this `outputAmount` is not correct. The test before the changes uses `swapIdToEgressAmount[gasSwapId]` to get the gas output amount. However, here the `outputAmount` comes from the SwapExecuted with swapRequestId that is the principal swap, isn't it? We should be getting that from the SwapExecuted that is the gas swap instead.